oidc-service-impl/src/main/java/io/mosip/esignet/services/ConsentHelperService.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package io.mosip.esignet.services;
+
+import com.auth0.jwt.interfaces.Claim;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.JWTClaimsSet;
+import io.mosip.esignet.api.dto.ClaimDetail;
+import io.mosip.esignet.api.dto.Claims;
+import io.mosip.esignet.api.util.ConsentAction;
+import io.mosip.esignet.core.constants.ErrorConstants;
+import io.mosip.esignet.core.dto.*;
+import io.mosip.esignet.core.exception.EsignetException;
+import io.mosip.esignet.core.exception.InvalidTransactionException;
+import io.mosip.esignet.core.spi.ConsentService;
+import io.mosip.esignet.core.util.IdentityProviderUtil;
+import io.mosip.esignet.core.util.KafkaHelperService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.text.ParseException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static io.mosip.esignet.core.util.IdentityProviderUtil.ALGO_SHA3_256;
+
+@Slf4j
+@Component
+public class ConsentHelperService {
+
+    @Autowired
+    private CacheUtilService cacheUtilService;
+
+    @Autowired
+    private ConsentService consentService;
+
+    private final ObjectMapper objectMapper=new ObjectMapper();
```

Comment:
ObjectMapper instance is available in the context, you can auto wire objectMapper

Reply:
removed unused objectMapper